cartographer_rviz/cartographer_rviz/submaps_display.cc

```diff
@@ -149,6 +164,51 @@ void SubmapsDisplay::update(const float wall_dt, const float ros_dt) {
   }
 }
 
+void SubmapsDisplay::AllEnabledChanged() {
+  ::cartographer::common::MutexLocker locker(&mutex_);
+  const bool visibility = visibility_all_enabled_->getBool();
+  for (auto& trajectory : trajectories_) {
+    for (auto& submap : trajectory) {
+      DisconnectVisibilityChange(submap);
+      submap->set_visibility(visibility);
+      ConnectVisibilityChange(submap);
+    }
+  }
+}
+
+void SubmapsDisplay::VisibilityChanged(DrawableSubmap* const submap) {
+  ::cartographer::common::MutexLocker locker(&mutex_);
+  // Handle the case when the user was holding Ctrl,
+  // when we also toggle the visibility of neighbouring submaps
+  auto modifiers = ::QApplication::keyboardModifiers();
+  if (modifiers.testFlag(::Qt::ControlModifier)) {
+    const bool visibility = submap->visibility();
+    for (int i = -1; i < 2; i += 2) {
+      try {
+        auto& neighbour_submap = trajectories_[submap->trajectory_id()].at(
+            static_cast<size_t>(submap->submap_index() + i));
+        DisconnectVisibilityChange(neighbour_submap);
+        neighbour_submap->set_visibility(visibility);
+        ConnectVisibilityChange(neighbour_submap);
+      } catch (std::out_of_range& err) {
+      }
+    }
+  }
+}
+
+void SubmapsDisplay::DisconnectVisibilityChange(
+    const std::unique_ptr<DrawableSubmap>& submap) {
```

Comment:
Nit: I would prefer to take the `const DrawableSubmap* const` ptr directly instead of the reference to the unique_ptr.